src/tensorflow_encrypted/protocol/pond.py

```diff
@@ -1263,24 +1293,107 @@ def _dot_masked_masked(prot, x, y):
 
     with tf.name_scope('dot'):
 
-        with tf.device(self.crypto_producer.device_name):
+        with tf.device(prot.crypto_producer.device_name):
             ab = a.dot(b)
             ab0, ab1 = _share(ab)
 
-        with tf.device(self.server_0.device_name):
+        with tf.device(prot.server_0.device_name):
             alpha = alpha_on_0
             beta = beta_on_0
             z0 = ab0 + a0.dot(beta) + alpha.dot(b0) + beta.dot(alpha)
 
-        with tf.device(self.server_1.device_name):
+        with tf.device(prot.server_1.device_name):
             alpha = alpha_on_1
             beta = beta_on_1
             z1 = ab1 + a1.dot(beta) + alpha.dot(b1)
 
+
     z = PondPrivateTensor(z0, z1)
     z = prot.truncate(z)
     return z
 
+#
+# Conv helpers
+#
+# TODO[koen] create operations for all possible combinations
+
+def _conv2d_int100(x, y, strides, padding="SAME"):
```

Comment:
seems to make sense to move this to the backing tensor..?

Reply:
yes definitely

Reply:
done